app/interpreters/message_interpreter.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class MessageInterpreter
+  attr_reader :scorecard, :message
+
+  def initialize(scorecard_id, message)
+    @scorecard = ::Scorecard.find_by id: scorecard_id
+    @message = message
+  end
+
+  def interpreted_message
+    return message if scorecard.nil?
+
+    sms = message
+    codes.each do |code|
```

Comment:
@hengsokly should use rename to use variable instead of code

embeded_variables. each do |variable|
  ...
end